site/docs/v1/tech/events-webhooks/signing.adoc

```diff
@@ -0,0 +1,108 @@
+---
+layout: doc
+title: Signing Webhooks
+description: Learn how to sign your Webhook events so you can verify they were sent by FusionAuth
+navcategory: developer
+---
+
+== Signing Webhooks
+
+Signing Webhook events allows you to verify each event was sent by FusionAuth.  Webhook signatures provide an extra layer of security in addition to the methods described in link:/docs/v1/tech/events-webhooks/securing[Securing a Webhook].  Signature verification provides protection against bad actors spoofing webhook event payloads to look like they came from FusionAuth.
+
+This document covers configuring webhook signatures, verifying signatures in your webhook listener, and key rotation.
+
+* <<Configuration>>
+* <<Signature Verification>>
+* <<Example Webhook Listener Code>>
+* <<Testing>>
+* <<Key Rotation>>
+
+=== Configuration
+
+Configuring webhook signatures in FusionAuth consists of generating a key and configuring your webhook to sign using that key.
+
+image::webhook-key-gen.png[Webhook Key Generation,width=1200,role=shadowed]
+
+Keys are generated or imported from [breadcrumb]#Settings -> Key Master#. Webhooks can be signed with three types of keys
+
+* EC key - strongest cryptography, public key can be available
+* RSA key - strong cryptography, public key can be available
+* HMAC key - fast cryptography, requires manual key distribution
+
+EC and RSA keys allow you to make public keys available through the `/.well-known/jwks.json` endpoint, which facilitates key rotation. If your webhook listener cannot make outbound network connections or you prefer to manually configure your key in your webhook listener, HMAC keys are a good option.
+
+For this example, we'll generate an RSA key pair.
```

Comment:
There aren't details on generating the RSA key pair. It seems like it just jumps from "let's create an RSA key pair" straight to assigning it.

Reply:
I don't think additional details are really necessary in this guide, but added a link to the Key Master Guide.